qa/tasks/cephfs/test_scrub_checks.py

```diff
@@ -443,6 +444,50 @@ def test_scrub_split_dirfrags(self):
             timeout=30
         )
 
+    def test_scrub_backtrace_divergent_newer(self):
+        """
+        The consequence of the scrub will not acquire nest locks which would update
+        the iversion for the in-memory inode prior to comparing to on-disk values.
+        """
+
+        self.fs.set_max_mds(2)
+        self.fs.wait_for_daemons()
+
+        test_path2 = "testdir1/testdir2"
+        test_path3 = f"{test_path2}/testdir3"
+        test_path4 = f"{test_path3}/testdir4"
+        test_file = f"{test_path3}/file_0"
+
+        self.mount_a.run_shell(["mkdir", "-p", test_path4])
+        self.mount_a.setfattr(test_path2, "ceph.dir.pin", "0") # no surprise migrations
+
+        # Will trigger to load ancestor inodes in mds.1
+        self.mount_a.setfattr(test_path4, "ceph.dir.pin", "1")
+        self.mount_a.run_shell_payload(f"cd {test_path4} && for i in `seq 1 10`; do touch file_$i; done;")
+
+        # Dirty the ancestor inodes in mds.0 and then the mds.1 version will become older
+        self.mount_a.run_shell(["touch", f"{test_file}"])
+        self.mount_a.run_shell_payload(f"cd {test_path3} && for i in `seq 1 32`; do mkdir dir_$i; done;")
+
+        # Flush path 'testdir3/file_0' and this will persist the backtrace to disk
+        self.fs.rank_asok(["flush_path", f"/{test_file}"], rank=0)
+
+        # Export testdir3 to mds.1
+        self.fs.rank_asok(["export", "dir", f"/{test_path3}", "1"], rank=0)
+
+        # Scrub 'testdir3/file_0' and it will detect the backtrace in memory will be older and
+        # metadata damage will be reported via cluster health warnings.
+        out_json = self.fs.run_scrub(["start", f"/{test_file}", "recursive"])
```

Comment:
I think you could do a `recursive,force` scrub on `/` and reproduce?

Reply:
If we only scrub on `/` also could be reproduced, but we must remove the `force` option instead. Else I couldn't reproduce it by trying many times.